soc/nordic/common/nrf_sys_event.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <nrf_sys_event.h>
+
+#if CONFIG_SOC_SERIES_NRF54HX
+
+/*
+ * The 54HX is not yet supported by an nrfx driver nor the system controller so
+ * we implement an ISR and concurrent access safe reference counting implementation
+ * here using the nrfx hal.
+ */
+
+#include <hal/nrf_lrcconf.h>
+
+static struct k_spinlock global_constlat_lock;
+static uint16_t global_constlat_count;
+
+int nrf_sys_event_request_global_constlat(void)
+{
+	K_SPINLOCK(&global_constlat_lock) {
+		if (global_constlat_count == 0) {
+			nrf_lrcconf_task_trigger(NRF_LRCCONF000,
+						 NRF_LRCCONF_TASK_CONSTLAT_ENABLE);
+			nrf_lrcconf_task_trigger(NRF_LRCCONF010,
+						 NRF_LRCCONF_TASK_CONSTLAT_ENABLE);
+#if CONFIG_SOC_NRF54H20_CPURAD
+			nrf_lrcconf_task_trigger(NRF_LRCCONF020,
+						 NRF_LRCCONF_TASK_CONSTLAT_ENABLE);
+#endif
+			while (!nrf_lrcconf_constlatstat_check(NRF_LRCCONF000)) {
+			}
```

Comment:
This is different behaviour vs what is done for other SoCs - `nrfx_power_constlat_mode_request` only triggers a task and does not wait.
If that is expected then it should be mentioned in docs at least.

Reply:
the docs just state that it can be checked, maybe that means the task is "atomic"?

Reply:
removed the check :)